store/storage/rocksdb/iterator.go

```diff
@@ -0,0 +1,129 @@
+package rocksdb
+
+import (
+	"bytes"
+
+	"cosmossdk.io/store/v2"
+	"github.com/linxGnu/grocksdb"
+)
+
+var _ store.Iterator = (*iterator)(nil)
+
+type iterator struct {
+	source             *grocksdb.Iterator
+	prefix, start, end []byte
+	reverse            bool
+	invalid            bool
+}
+
+func newRocksDBIterator(source *grocksdb.Iterator, prefix, start, end []byte, reverse bool) *iterator {
+	if reverse {
+		if end == nil {
+			source.SeekToLast()
+		} else {
+			source.Seek(end)
```

Comment:
What happens if `end = make([]byte, 0)` and not `end = []byte(nil)` https://go.dev/play/p/5bY47B5BITP?
Then what will happen to `source.Seek([]byte{})`